src/app/find/errorReports/ErrorReportContainer.jsx

```diff
@@ -0,0 +1,187 @@
+/* eslint-disable max-len */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router';
+import { Route, Redirect, Switch, BrowserRouter as Router } from 'react-router-dom';
+import { postErrorReport, getLocation } from '../../../actions';
+import { selectLocationError, selectLocationData } from '../../../selectors/location';
+import LoadingLabel from '../../../components/form/LoadingLabel';
+import NotFound from '../../notFound/NotFound';
+import ErrorLabel from '../../../components/form/ErrorLabel';
+import Modal from '../../../components/modal';
+import Icon from '../../../components/icon';
+import ErrorReportText from './ErrorReportText';
+import ErrorReportInformationSelect from './ErrorReportInformationSelect';
+import Thanks from './Thanks';
+
+const LoadingView = ({ message }) => (
+  <div className="m-4">
+    <LoadingLabel>
+      {message}
+    </LoadingLabel>
+  </div>
+);
+
+class ErrorReportContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      errorReportInformation: {
+        errorReportGeneralLocationError: false,
+        errorReportServices: [],
+      },
+      errorReportText: '',
+    };
+
+    this.onErrorReportGeneralLocationErrorChanged = this.onErrorReportGeneralLocationErrorChanged.bind(this);
+    this.onErrorReportTextChanged = this.onErrorReportTextChanged.bind(this);
+    this.onErrorReportServicesChanged = this.onErrorReportServicesChanged.bind(this);
+    this.onErrorReportSubmitted = this.onErrorReportSubmitted.bind(this);
+  }
+
+  componentDidMount() {
+    if (!this.props.locationData) {
+      this.props.getLocation();
+    }
+  }
+
+  onErrorReportTextChanged(event) {
+    this.setState({ errorReportText: event.target.value });
+  }
+
+  onErrorReportServicesChanged(value) {
+    const newErrorReportInformation = { ...this.state.errorReportInformation };
+
+    const newErrorReportServices = newErrorReportInformation.errorReportServices.includes(value)
```

Comment:
This is the kind of low-level logic I'd keep to the specific pages' components. Otherwise, this container doesn't just concern itself with the high-level flow, but basically has all the logic for every level of the flow in it.
Ideally, its function would just be:
```
onErrorReportServicesChanged(errorReportInformation) {
  this.setState({ errorReportInformation });
}
```
And the child components would make sure to do everything else.

But if it's not a really quick fix, I wouldn't bother with it right now.